### PR TITLE
New location UCR auto-filters

### PR DESCRIPTION
**Pull request by kaapstorm**
Still need to add tests. PRing because I had a couple of questions.

[FB 215658](http://manage.dimagi.com/default.asp?215658)

(Sneaking in a "this month" date range filter, requested by Nick Nestle, which fits in nicely with the next UCR filter I need to add ([FB 215656](http://manage.dimagi.com/default.asp?215656)) ... although, as I write this, I'm thinking I'll cherrypick it across when I PR that. Feel free to ignore.)

@esoergel @nickpell cc @NoahCarnahan @snopoke

### CODE REVIEW

File: corehq/apps/reports/daterange.py [hunk 1]
```diff
@@ -1,7 +1,7 @@
 from collections import namedtuple
 import datetime
 from django.utils.translation import ugettext_lazy as _
-from corehq.apps.accounting.utils import get_previous_month_date_range
+from corehq.apps.accounting.utils import get_previous_month_date_range, get_current_month_date_range
```
Comment (czue) [inline corehq/apps/reports/daterange.py]: nit: seems a bit weird for these to be in `accounting.utils`

**Comment by czue** [comment]
looks good! don't think any major blockers on my side but will let you get to where you're happy with it

**Comment by kaapstorm** [comment]
(Failure real. On it.)

File: corehq/apps/app_manager/tests/test_filters.py [hunk 2]
```diff
@@ -0,0 +1,167 @@
+from datetime import datetime
+
+from django.test import SimpleTestCase, TestCase
```
Comment (dimagimon) [inline corehq/apps/app_manager/tests/test_filters.py]: F401 'SimpleTestCase' imported but unused

**Comment by kaapstorm** [comment]
Fixed merge conflicts

**Comment by snopoke** [comment]
:mans_shoe: :construction_worker: :dollar: test :x:

**Comment by czue** [comment]
:dollar: test makes me :rage:

**Comment by kaapstorm** [comment]
Aaargh! Error is the result of merge (importing datetime not datetime.datetime). On it.

**Comment by kaapstorm** [comment]
:mans_shoe: :construction_worker: :dollar: :recycle:

**Comment by benrudolph** [comment]
sometimes i see a future where there are no words

**Comment by esoergel** [comment]
:rocket: :no_mouth: :rocket:

**Comment by kaapstorm** [comment]
:exclamation: OMG!!1! :exclamation: :construction_worker: :dollar: :white_check_mark: :exclamation:

**Comment by kaapstorm** [comment]
@benrudolph, there is evidence to suggest that severing the two hemispheres in the brain ("corpus callosotomy") results in [two consciousnesses][2] in a single person! (Ca-ray-zee, right?!) Also, the [brain-to-brain interface][1] is a thing. (Not the right kind of thing, but a step in that direction. And of course the split consciousness thing is controversial and has alternative explanations, so quite possibly BS.)

BUT maybe, just maybe, you can connect the hemispheres of separate people's brains, and result in a single consciousness: One person thinking with their own left hemisphere and someone else's right hemispere. Or multiple people's right hemispheres.

Who needs words?

(Of course it might not be possible, and all this is ridiculous, but fun to consider. There's a great sci-fi story in there somewhere.)


  [1]: http://www.extremetech.com/extreme/165081-first-human-brain-to-brain-interface-allows-remote-control-over-the-internet-telepathy-coming-soon

  [2]: https://en.wikipedia.org/wiki/Dual_consciousness_%28neuroscience%29

**Comment by czue** [comment]
lol. i bet brain to brain communication looks like this: :cat2: :cat2:

**Comment by kaapstorm** [comment]
LOL.

Or [Rat Neo](http://www.bbc.com/news/science-environment-21604005).

**Comment by kaapstorm** [comment]
Any chance of getting this merged so I can deploy to India and Matt can play with it before he goes on leave?

**Pull request merged**